packages/dds/tree/src/feature-libraries/editable-tree/editableTreeContext.ts

```diff
@@ -82,6 +99,39 @@ export interface EditableTreeContext {
      * is committed successfully.
      */
     attachAfterChangeHandler(afterChangeHandler: (context: EditableTreeContext) => void): void;
+
+    /**
+     * Types an arbitrary data with the given `TreeSchemaIdentifier` and wraps it into a format,
+     * which can be accepted by the EditableTree when using simple assignments.
+     *
+     * Use it whenever the field is polymorphic, to explicitly define the type of your data.
+     */
+    newDetachedNode<T extends Brand<any, string>>(
+        type: TreeSchemaIdentifier,
+        data: T extends BrandedType<infer ValueType, infer Name>
+            ? BrandedType<ValueType, Name>
+            : never,
+    ): T;
+    /**
+     * Types an arbitrary data with the given `TreeSchemaIdentifier` and wraps it into a format,
+     * which can be accepted by the EditableTree when using simple assignments.
+     *
+     * Use it whenever the field is polymorphic, to explicitly define the type of your data.
+     */
+    newDetachedNode<T extends Brand<any, string>>(
+        type: TreeSchemaIdentifier,
+        data: T extends BrandedType<infer ValueType, string> ? ValueType : never,
+    ): T;
+    /**
+     * _Type-unsafe overload of the method allowing to change EditableTrees using simple assignments in contexts,
+     * where proper typing is not possible or not important._
+     *
+     * Types an arbitrary data with the given `TreeSchemaIdentifier` and wraps it into a format,
+     * which can be accepted by the EditableTree when using simple assignments.
+     *
+     * Use it whenever the field is polymorphic, to explicitly define the type of your data.
+     */
+    newDetachedNode<T extends Brand<any, string>>(type: TreeSchemaIdentifier, data: unknown): T;
```

Comment:
This is the most tricky part for me. It allows to use a very specific typing as well as no typing at all. However, if one misses to properly type a part of the data (like missing the type where it must be), the whole type check is discarded as the least type-safe overload will be used. So, not sure at all, how to deal with it.
There are a lot of examples in `editableTree.editing.spec.ts` with regards.